From d9ff7a6f8050bee5453d6fe62cf51779596a47ce Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Fri, 22 Nov 2024 10:31:07 -0800
Subject: [PATCH] configure: Enforce 4-byte stack alignment on clang for PE
 builds.

Same as -mpreferred-stack-boundary=2 for gcc.
Clang interprets this as "n" bytes, while gcc uses "2^n" bytes.
https://linux.kernel.narkive.com/y7hAplfS/patch-x86-build-specify-stack-alignment-for-clang

(cherry picked from commit c4f680a24da29233744b52aa3e08196f5ad05a1b)
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index c54f048a71d..2e85f1e1814 100644
--- a/configure.ac
+++ b/configure.ac
@@ -582,6 +582,7 @@ This is an error since --enable-archs=$wine_arch was requested.])])
 
     case $wine_arch in
       i386) WINE_TRY_PE_CFLAGS([-fno-omit-frame-pointer])
+            WINE_TRY_PE_CFLAGS([-mstack-alignment=4])
             WINE_TRY_PE_CFLAGS([-mpreferred-stack-boundary=2])
             WINE_TRY_PE_CFLAGS([-Wl,--disable-stdcall-fixup],
                                [AS_VAR_APPEND([${wine_arch}_LDFLAGS],[" -Wl,--disable-stdcall-fixup"])]) ;;
